README.md

```diff
@@ -3,33 +3,161 @@
+
+Use this README to understand the purpose, basic usage, and common configurable values for PDK templates. For further information about PDK, refer to our [documentation site](https://www.puppet.com/docs/pdk/2.x/pdk.html).
 
+#### Table of contents
+
+* [Introduction](#pdk-templates)
+* [How templates are used by PDK](#how-templates-are-used-by-pdk)
 * [Basic usage](#basic-usage)
-* [Values of config\_defaults](#values)
-* [Making local changes to the Template](#making-local-changes-to-the-template)
-* [Further Notes](#notes)
+* [Adjusting templates via .sync.yml](#adjusting-templates-via-syncyml)
+  * [Adding configuration values](#adding-configuration-values)
+  * [Removing default configuration values](#removing-default-configuration-values)
+  * [Excluding files from default behaviour](#excluding-files-from-default-behaviour)
+  * [Unmanaged and delete keys](#unmanaged-and-delete-keys)
+  * [Setting custom gems in the Gemfile](#setting-custom-gems-in-the-gemfile)
+  * [Manage Rubocop rules](#manage-rubocop-rules)
+* [PDK default config values](#pdk-default-config-values)
+* [Further Notes](#further-notes)
+
+## How templates are used by PDK
+
+When you use PDK to build, convert, or update a module, PDK accesses the relevant template in the pdk-templates repository to retrieve default “instructions” on how to build the module. Those instructions are applied to the local repo in which the command was run. 
+
+When configuring the module, PDK reads a set of default settings in `.config_defaults.yml` and merges it with the configuration settings in `.sync.yml`.
+
+The result of this merge creates a hidden hash containing the final configuration of the module that PDK applies. Top-level keys of this resulting hash correspond to target files. A global configuration is merged with the configuration hash for a particular target file. This allows module developers to override or amend the configuration by adding new values through `.sync.yml`. For example, a knockout prefix (---) can be applied to keys in `.sync.yml` to remove default configuration values. 
+
+Target files are created by rendering a corresponding template which refers its configuration via the `@configs` hash.
+
+You can configure PDK to make files executable if the file is executable in the template. To do this, include `manage_execute_permissions: true` either in the target file or globally via the common config key.
 
 ## Basic Usage
 
-Templates like this one can be used in conjunction with the PDK. As default the PDK itself uses the templates within this repository to render files for use within a module. Templates can be passed to the PDK as a flag for several of the commands.
+By default PDK uses the templates within this repository to render files for use within a module. To use a third-party template, pass its URL using the `–-template-url` flag as shown in the following example: 
```

Comment:
Should we note that the PDK will look for a branch called `main` unless specified? The error provided if that branch doesn't exist isn't clear that the branch couldn't be found, e.g.
```
pdk (FATAL): Unable to find a valid template at https://github.com/org/repo.git#main
```

Reply:
I feel like it should be expected that the command would default to main. But I guess a wee note won't hurt. Specially if there is people out there still using master in their naming conventions.